File: nestjs/src/controllers/decode.controller.ts
import {
  Body,
  Controller,
  NotImplementedException,
  Post,
} from '@nestjs/common';
import { ShortenedUrlDto } from '../dtos/shortenedUrl.dto';
import { UrlDto } from '../dtos/Url.dto';

@Controller()
export class DecodeController {
  @Post('/decode')
  decode(@Body() shortenedUrlDto: ShortenedUrlDto): UrlDto {
    throw new NotImplementedException();
  }
}

File: dotnet/UrlShortener/Controllers/DecodeController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using UrlShortener.Dtos;

namespace UrlShortener.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class DecodeController : ControllerBase
    {
        private readonly ILogger<DecodeController> _logger;

        public DecodeController(ILogger<DecodeController> logger)
        {
            _logger = logger;
        }

        [HttpPost]
        public UrlDto Decode(ShortenedUrlDto request) => throw new System.NotImplementedException();
    }
}
